src/references/script_debugger.md

```diff
@@ -0,0 +1,100 @@
+
+!!! info "Script Debugger and External Editors"
+    The **Script Debugger** only works with the **Script Editor** in **Core**, and not external editors.
+
+![Opening](../img/ScriptDebugger/opening_script_debugger.png)
+![Window](../img/ScriptDebugger/script_debugger_window.png)
+{: .image-cluster}
+
+## Enabling the Script Debugger
+
+The **Script Debugger** needs to be enabled so any breakpoints added, will pause the execution of the game. This allows areas of interest in a script to be inspected that may contain an issue.
+
+Clicking on the ![Disabled](../img/ScriptDebugger/script_debugger_disabled.png)icon will enable the **Script Debugger**. When enabled, the icon will turn red ![Enabled](../img/ScriptDebugger/script_debugger_enabled.png), indicating the **Script Debugger** is turned on. With the **Script Debugger** now turned on, any breakpoints added will pause execution of the game.
+{: .image-inline-text .image-background }
+
+!!! info "When disabling the **Script Debugger**, all breakpoints will also be disabled. This is useful, because all breakpoints don't need to be removed. Next time the **Script Debugger** is enabled, the breakpoints will also be enabled."
+
+## Enabling Pause on Error
+
+The **Script Debugger** has the option to automatically pause the game when an error has occurred. This can be useful to have enabled when scripts in a game are causing an error. Clicking on the ![Disabled](../img/ScriptDebugger/pause_on_error_disabled.png) icon will change to red ![Enabled](../img/ScriptDebugger/pause_on_error_enabled.png), indicating that the **Script Debugger** will pause on error.
+{: .image-inline-text .image-background }
+
+## Adding and Removing Breakpoints
+
+A breakpoint is a marker that can set for a specific line in a script. More than one marker can be set in a script, but only one can be set for each line. Markers can be set by left clicking with the mouse in the far left margin of the **Script Editor**.
+
+### Adding a Breakpoint
+
+A breakpoint can be added so the **Script Debugger** knows where to pause execution. Adding a breakpoint can be done by left clicking with the mouse in the far left margin of the **Script Editor**. On clicking in the margin, a red circle will be added to the margin, indicating that line has a breakpoint.
+
+![!Add Breakpoint](../img/ScriptDebugger/add_breakpoint.png){: .center loading="lazy" }
+
+### Removing a Breakpoint
+
+Removing a breakpoint works the same as adding a breakpoint. Clicking on the red circle in the far left margin of **Script Editor** will remove the breakpoint. When the game is played, the **Script Debugger** will not pause execution if no breakpoints are present.
+
+## Stepping Through a Script
+
+Stepping through a script is an important part of debugging, it allows creators to walk through each line of the script to inspect the variables and objects to see what is contained in them.
```

Comment:
Cut "is an important part of debugging". 

Also, "walk through" just continues the "step" metaphor. It allows creators to ... 

okay I'm really not sure if it actually only executes one line at a time, or if it is just a way of shifting focus to inspect variables. But either way, avoid any movement metaphors and try to reframe it in terms of what information is available and what explicitly happens in terms of the editor.